### PR TITLE
Adding Classes for the Portfolio Board endpoints

### PR DESCRIPTION
**Pull request by AlanHolmes**
Added the classes for the Portfolio Board endpoints:
https://developer.teamwork.com/projects/portfolio-boards/boards-in-portfolio-view

There are insert/update/delete/get methods for all classes, even though they are not on the API documentation (I've let teamwork know about this), but they do work.

**Portfolio\Board**
- **getAll** /portfolio/boards/ - returns all the portfolio boards
- **get** /portfolio/boards/{boardId} - returns the specific board
- **insert** /portfolio/boards - creates a new board
- **update**  /portfolio/boards/{boardId} - updates the given board
- **delete** /portfolio/boards/{boardId} - deletes the given board

**Portfolio\Column**
- **getAllForBoard** /portfolio/boards/{boardId}/columns - returns all the columns for the given board
- **get** /portfolio/columns/{icolumnId} - returns the specific column
- **insert** /portfolio/boards/{boardId}/columns - adds a column to the given board
- **update**  /portfolio/columns/{columnId} - updates the given column
- **delete** /portfolio/columns/{columnId} - deletes the given column

**Portfolio\Card**
- **getAllForColumn** /portfolio/columns/{columnId}/cards - returns all the cards for the given column
- **get** /portfolio/cards/{cardId} - returns the specific card
- **insert** /portfolio/columns/{columnId}/cards - Adds a project to the given board (thus creating a card)
- **update** /portfolio/cards/{cardId}/move - Used to move a card from one column to another
- **delete** /portfolio/cards/{cardId} - deletes the given card (removes the project from the column)

### CODE REVIEW

File: src/Portfolio/Card.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                'attributes' => ['type' => 'integer'],
+            ],
+        ];
+    }
+
+    /**
+     * Get all the Columns for a Portfolio Column
+     * GET /portfolio/columns/{columnId}/cards
+     *
+     * @param integer $columnId
```
Comment (simonschaufi) [inline src/Portfolio/Card.php]: please use the short version `int`

File: src/Portfolio/Card.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+     *
+     * @return \TeamWorkPm\Response\Model
+     * @throws \TeamWorkPm\Exception
+     */
+    public function getAllForColumn($columnId)
+    {
+        // Doesnt work
```
Comment (simonschaufi) [inline src/Portfolio/Card.php]: what exactly doesn't work? If it doesn't work, why is this method implemented?

File: tests/Portfolio/CardTest.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
+<?php
+
+class CardTest extends TestCase
+{
+    /** @var \TeamWorkPm\Portfolio\Card */
+    private $model;
+
+    /** @var integer */
```
Comment (simonschaufi) [inline tests/Portfolio/CardTest.php]: please use the short version `int`

File: src/Portfolio/Card.php (hunk 2 above)
Comment (AlanHolmes) [reply src/Portfolio/Card.php]: Whoops, this comment was left by mistake.

Added it as a not to myself that it wasn't returning cards, was going to point this out to TW, but upon further investigation, it needed these changes to return the data: https://github.com/loduis/teamwork.com-project-management/pull/45/files#diff-3d898e219b939f5a7a7f0f84ab366b7cR78

I'll remove this comment

**Comment by AlanHolmes** [comment]
@simonschaufi made those changes 😄

**Pull request merged**